components/controller/src/identifier_controller.rs

```diff
@@ -413,6 +417,26 @@ impl IdentifierController {
         Ok(group_prefix)
     }
 
+    pub async fn notify_witnesses(&self) -> Result<usize, ControllerError> {
+        let mut n = 0;
+        let evs = self
+            .source
+            .partially_witnessed_escrow
+            .get_partially_witnessed_events();
+        for ev in evs {
+            // Elect the leader
+            // Leader is identifier with minimal index among all participants who
+            // sign event. He will send message to witness.
+            let min_idx = ev.signatures.iter().map(|at| at.index).min();
+            if min_idx == Some(0) {
```

Comment:
This comparison is problematic in the case of group events. There may be no signature of index 0, but they're enough of them to accept the event. How about `get_index` function instead of 0?